core/templates/dev/head/components/state-editor/state-editor-properties-services/state-editor.service.ts

```diff
@@ -84,13 +84,10 @@ export class StateEditorService {
     this.stateEditorDirectiveInitialised = true;
   }
 
-  isStateEditorInitialised(): boolean {
+  checkEventListenerRegistrationStatus(): boolean {
```

Comment:
Can you explain the rationale for why some of the original booleans are left intact, whereas others are deleted?

Reply:
This function was initially written to ensure that event listeners associated with the components have registered before 'stateEditorInitialized' is triggered. However, I observed that the hints, solution and content editors do not have the relevent event listeners in their directives so it's unnecessary to have them here.

Only the interaction, responses and the state editor directives have the relevent event listeners. This is why I've left these here.

A serious issue I noticed was that the multiple choice interaction does not have "solutions" so this method would always return false (stateSolutionEditorInitialised == false)! This would cause the editor to appear broken because we watch for the changes in this function to trigger 'stateEditorInitialized'. So we need to fix this ASAP!